torch/csrc/jit/mobile/debug_info.cpp

```diff
@@ -82,20 +85,21 @@ std::pair<std::string, std::string> getStackTraceWithModuleHierarchy(
   std::vector<StackEntry> stack_entries;
   std::string module_info =
       root_scope_string + "(" + top_module_type_name + ")";
-  std::string caller_fn_name = "FunctionName_UNKNOWN";
+  std::string caller_fn_name = "<forward>";
```

Comment:
For hardcoded to <forward>: This is because we use debug handles for symbolication. So debug handle correspond to some node in a graph like:
```
graph(%self, %in):
%x = getAttr...
%1 = aten::add
```
The graph may correspond to some function name `my_forward`. And in this case debug handle for %1 does not know that it is for `my_forward`. So we dont have access to that info.

I think this can be fixed but maybe in a separate PR.

> and then immediately appended (mainly, why weren't we appending it before --are we potentially doing it twice, given that above we also added module_info.append("::").append(prev_function_name))?

Good observation, but we are not doing that twice because `caller_fn_name` is passed to `getStackTraceWithModuleHierarchy` but in that function when we do `module_info.append("::").append(prev_function_name);`, that happens after `prev_function_name` updated from the `InlinedCallStack` corresponding to debug handle. That will be different from `caller_fn_name` because as you can see in, https://github.com/pytorch/pytorch/pull/62419/files#diff-fbfe49a9e94202548c66b89039d024a473f8fbf2d0c9f42a54cfe7a1d08499dfL52-L58, `prev_function_name` gets updated.

Reply:
Also I will update function name from `<forward>` to `unknown`